### PR TITLE
cache-agent for yurthub support '*'

### PR DESCRIPTION
**Pull request by hhstu**
Signed-off-by: 刘畅 <basefly@outlook.com>

fix https://github.com/openyurtio/openyurt/issues/722

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/yurthub/cachemanager/cache_manager.go [hunk 1]
```diff
@@ -570,7 +570,7 @@ func (cm *cacheManager) CanCacheFor(req *http.Request) bool {
 		// request with Edge-Cache header, continue verification
 	} else {
 		cm.RLock()
-		if !cm.cacheAgents.Has(comp) {
+		if !cm.cacheAgents.HasAny("*", comp) {
```
Comment (rambohe-ch) [inline pkg/yurthub/cachemanager/cache_manager.go:573]: how about add a unit test case for CanCacheFor func?

Comment (hhstu) [reply pkg/yurthub/cachemanager/cache_manager.go:573]: 这地方加单元测试看起来还挺麻烦，需要引入一些 moke 框架去打桩

Comment (rambohe-ch) [reply pkg/yurthub/cachemanager/cache_manager.go:573]: @hhstu maybe you need to add only a test case in `TestCanCacheFor` func. you can reference test code here:
https://github.com/openyurtio/openyurt/blob/master/pkg/yurthub/cachemanager/cache_manager_test.go#L2373-L2636

File: pkg/yurthub/cachemanager/cache_manager_test.go [hunk 2]
```diff
@@ -29,6 +30,13 @@ import (
 	"testing"
 	"time"
 
+	"github.com/openyurtio/openyurt/pkg/projectinfo"
+	hubmeta "github.com/openyurtio/openyurt/pkg/yurthub/kubernetes/meta"
```
Comment (Congrool) [inline pkg/yurthub/cachemanager/cache_manager_test.go:34]: We should import package in such format
```go
	"fmt"
	"net"
	"strings"
	"time"

	"github.com/spf13/pflag"
	"k8s.io/client-go/informers"
	"k8s.io/klog/v2"
	"sigs.k8s.io/apiserver-network-proxy/pkg/server"

	"github.com/openyurtio/openyurt/cmd/yurt-tunnel-server/app/config"
	"github.com/openyurtio/openyurt/pkg/projectinfo"
	"github.com/openyurtio/openyurt/pkg/yurttunnel/constants"
	kubeutil "github.com/openyurtio/openyurt/pkg/yurttunnel/kubernetes"
	"github.com/openyurtio/openyurt/pkg/yurttunnel/pki"
```
First block should be golang official pkg
Second block should be the 3-rd part pkg
Third block should be the openyurt pkg

You can find more details at #694

Comment (hhstu) [reply pkg/yurthub/cachemanager/cache_manager_test.go:34]: get

File: pkg/yurthub/cachemanager/cache_manager.go (hunk 1 above)
Comment (hhstu) [reply pkg/yurthub/cachemanager/cache_manager.go:573]: down

**Review by rambohe-ch — APPROVED** [review]

**Pull request merged**